Allow to assumeRole from ConfigurationProvider (#803)
async-aws/aws: docs/configuration.md

### roleArn

> The Amazon Resource Name (ARN) of the role that the caller is assuming.

Reviewer: ```suggestion
The Amazon Resource Name (ARN) of the role that the client should be "assuming" after authentication. 
```
